### DISCUSSION TITLE
How to know if a model is a user or an admin?

### DISCUSSION THREAD

**Discussion by heloineto** [pocketbase/pocketbase#3122] [Q&A] [answered]
Is there a way to known if the current logged in entity is a user or an admin?

Example:

```js
const { model } = pb.authStore;
const isAdmin = /* What goes here? */
```

I would really appreciate it if there was a default way of differentiating the two, without making any modifications to the tables

**Comment by ganigeorgiev** [comment, marked as answer]
With the next JS SDK version there will be helper getters (eg. isAdmin) to determine the model type, but for now you generally have 3 options:

1. Check for the existence of a field that only the auth records have, eg. `collectionId`:
    ```js
    import PocketBase, { Admin } from "pocketbase"

    const pb = new PocketBase(url);

    const isAdmin = pb.authStore.model && !pb.authStore.model.collectionId;
    ```

2. Parse the token and check its type:
    ```js
    import PocketBase, { getTokenPayload } from "pocketbase"

    const pb = new PocketBase(url);

    const payload = getTokenPayload(pb.authStore.token);

    console.log(payload.type === "admin") // admin or authRecord
    ```
    
3. Check the model instance (**this will not work with the next release since the custom classes will be removed!**):
    ```js
    import PocketBase, { Admin } from "pocketbase"

    const pb = new PocketBase(url);

    console.log(pb.authStore.model instanceof Admin)
    ```

**Reply by heloineto** [reply]
Thank you so much, you're amazing!
